tasks/prepare-fbc-release/prepare-fbc-release.yaml

```diff
@@ -0,0 +1,114 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: prepare-fbc-release
+  labels:
+    app.kubernetes.io/version: "1.0.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task for prepration of FBC release this task will
+    collect components from snapshot and set OCP version by
+    inspecting containerImage for each component.
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the Snapshot spec in the data workspace
+      type: string
+      default: "snapshot_spec.json"
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+      default: "data.json"
+  workspaces:
+    - name: data
+      description: Workspace where the snapshot is stored
+  steps:
+    - name: prepare-fbc-release
+      image: quay.io/hacbs-release/release-utils:a918b3a7cbd40add5955a2f1acb4c3dbb29c44e9
+      script: |
+        #!/usr/bin/env bash
+        set -eu
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+        if [ ! -f "${SNAPSHOT_PATH}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi        
+        
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        # Verify data.json exists
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+        fi        
+
+        pattern="^v[0-9]+\.[0-9]+$"
+
+        # Read components and initial values
+        components=$(jq -c '.components[]' "$SNAPSHOT_PATH")
+        fromIndex=$(jq -r '.fbc.fromIndex' "$DATA_FILE")
+        targetIndex=$(jq -r '.fbc.targetIndex' "$DATA_FILE")
+        binaryImage=$(jq -r '.fbc.binaryImage' "$DATA_FILE")
+
+        # Print initial values
+        echo "Initial fromIndex: $fromIndex"
+        echo "Initial targetIndex: $targetIndex"
+        echo "Initial binaryImage: $binaryImage"
+        echo
+
+        # Count components
+        COMPONENT_COUNT=$(jq '.components | length' "$SNAPSHOT_PATH")
+        echo "found $COMPONENT_COUNT components"
+
+        # Initialize jq_script
+        jq_script="."
+
+        # Function to replace tag in an image
+        replace_tag() {
+            local updatedImage="${1%:*}:${2}"
+            echo "$updatedImage"
+        }
+
+        # Iterate over components
+        while read -r component; do
+            containerImage=$(echo "$component" | jq -r '.containerImage')
+            componentName=$(echo "$component" | jq -r '.name')
+
+            # Fetch OCP version using skopeo
+            ocpVersion=$(skopeo inspect --raw docker://"$containerImage" \
+                | jq -r '.annotations."org.opencontainers.image.base.name"' | cut -d: -f2)
+
+            # Check if the version matches the pattern
+            if ! echo "$ocpVersion" | grep -Eq "$pattern"; then
```

Comment:
I would go instead for a positive logic instead. Also I would use `[[ "${ocpVersion}" =~ ^v[0-9]+\.[0-9]+$ ]]` instead. IMHO it reads better.

Reply:
I agree `[[ ~= ]]` would be nicer here. But not sure what you mean by positive logic, Leandro? Because we only want to skip it if it doesn't match it. So if you want to invert the logic, nothing would happen in the positive branch. Of course the whole rest of the commands could be in the `then` branch, but I don't think that would be pretty.

Reply:
I personally think  that it is easier to understand if you ask "is something equal other-thing" than "is something not equal to other-thing". Other than that, if the logic you described makes no sense, i am ok keeping the current way.

Reply:
But would still move to use`[[ ]]`